docs(azure): update azuredevops docs to match new changes (#33788)
renovatebot/renovate: lib/modules/platform/azure/readme.md

    inputs:
      workingFile: .npmrc

  - bash: |
      add-apt-repository ppa:git-core/ppa
      apt update && apt install git -y

>     displayName: 'Install latest git version'

Reviewer: ```suggestion
    displayName: 'Install latest version of Git'
```

Are you allowed to use capitalization in the `displayName` field/variable?

Author: Yup, done

For the `repositories` key, replace `YOUR-PROJECT/YOUR-REPO` with your Azure DevOps project and repository.

### Using Azure DevOps internal API for pipeline tasks versions

<!-- prettier-ignore -->
!!! info

>     Renovate now uses the set of APIs that Azure provides to query the azure-pipelines tasks versions directly from the instance. Read [pull request #32966](https://github.com/renovatebot/renovate/pull/32966) and [discussion #24820](https://github.com/renovatebot/renovate/discussions/24820) for more background information on this change.

Reviewer: ```suggestion
    Renovate now uses the set of APIs that Azure provides to query the azure-pipelines tasks versions directly from the instance. Read [pull request 32966](https://github.com/renovatebot/renovate/pull/32966) and [discussion 24820](https://github.com/renovatebot/renovate/discussions/24820) for more background information on this change.
```

The `#` characters are commenting out the line. Let's remove the `#` bits. This way we can see the full sentence when published. 😉

For the `repositories` key, replace `YOUR-PROJECT/YOUR-REPO` with your Azure DevOps project and repository.

### Using Azure DevOps internal API for pipeline tasks versions

<!-- prettier-ignore -->
!!! info

>     Renovate bot is now using the set of APIs that azure provides to query the azure-pipelines tasks versions directly from the instance. ([PR #32966](https://github.com/renovatebot/renovate/pull/32966) and [Discussion #24820](https://github.com/renovatebot/renovate/discussions/24820))

Reviewer: ```suggestion
    Renovate now uses the set of APIs that Azure provides to query the azure-pipelines tasks versions directly from the instance. Read [pull request #32966](https://github.com/renovatebot/renovate/pull/32966) and [discussion #24820](https://github.com/renovatebot/renovate/discussions/24820) for more background information on this change.
```

> To use it, you need to have in your config :

Reviewer: ```suggestion
To let Renovate use the Azure DevOps internal API, you must set these variables in your config:
```

- `platform` = `azure`

> - `endpoint` = `$(System.CollectionUri)` (an [azure predefined variable](https://learn.microsoft.com/en-us/azure/devops/pipelines/build/variables?view=azure-devops&tabs=yaml))

Reviewer: ```suggestion
- `endpoint` = `$(System.CollectionUri)`, this is an [Azure predefined variable](https://learn.microsoft.com/en-us/azure/devops/pipelines/build/variables?view=azure-devops&tabs=yaml)
```

- `hostRules.hostType` = `azure-pipelines-tasks`

> An example configuration file, in `.json` format this time, would be :
> 

Reviewer: ```suggestion
```

Let's use a codeblock title instead of this line. :wink: 

[Material for MkDocs, adding a title to a codeblock](https://squidfunk.github.io/mkdocs-material/reference/code-blocks/#adding-a-title)

> ```json

Reviewer: ```suggestion
```json title="Example config file in JSON format"
```

For the `repositories` key, replace `YOUR-PROJECT/YOUR-REPO` with your Azure DevOps project and repository.

### Using Azure DevOps internal API for pipeline tasks versions

!!! info
    Renovate bot is now using the set of APIs that azure provides to query the azure-pipelines tasks versions directly from the instance. ([PR #32966](https://github.com/renovatebot/renovate/pull/32966) and [Discussion #24820](https://github.com/renovatebot/renovate/discussions/24820))

    To use it, you need to have in your config :

      - `platform` = `azure`
      - `endpoint` = `$(System.CollectionUri)` (an [azure predefined variable](https://learn.microsoft.com/en-us/azure/devops/pipelines/build/variables?view=azure-devops&tabs=yaml))
      - `hostRules.hostType` = `azure-pipelines-tasks`

An example configuration file, in `.json` format this time, would be :

```json
{
  "platform": "azure",
  "endpoint": "https://dev.azure.com/ORG_NAME",

>   "token": "process.env.RENOVATE_TOKEN",

Reviewer 2: JSON files cannot read from `process.env`. This example would put the _literal string value_ `"process.env.RENOVATE_TOKEN"` into the file, and not the _value_ from env. Same as for the hostRules below

Author: Hey @rarkins , it works on our on-prem azuredevops server.

Would it then works just because I'm calling `npx renovate` with these env vars ?

      RENOVATE_PLATFORM: azure
      RENOVATE_ENDPOINT: $(System.CollectionUri)
      RENOVATE_CONFIG_FILE: $(Build.SourcesDirectory)/renovate_bot_config.json
      RENOVATE_TOKEN: $(System.AccessToken)
      LOG_LEVEL: debug

Author: I updated it.
